src/app/blog/[slug]/utils/getSharePostSocialIcons.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  FacebookLogo,
+  LinkedinLogo,
+  RedditLogo,
+  XLogo,
+} from '@phosphor-icons/react/dist/ssr'
+
+export const sharePostSocialIcons = [
+  {
+    icon: FacebookLogo,
+    label: 'Facebook',
+    href: `https://www.facebook.com/sharer/sharer.php?u=currentUrl&quote=postTitle`,
+  },
+  {
+    icon: LinkedinLogo,
+    label: 'LinkedIn',
+    href: `https://www.linkedin.com/shareArticle?mini=true&url=currentUrl&title=postTitle`,
+  },
+  {
+    icon: RedditLogo,
+    label: 'Reddit',
+    href: `https://reddit.com/submit?url=currentUrl&title=postTitle`,
+  },
+  {
+    icon: XLogo,
+    label: 'X',
+    href: `https://twitter.com/share?url=currentUrl&text=postTitle`,
+  },
+]
+
+export function getSharePostSocialIcons(currentUrl: string, postTitle: string) {
+  return sharePostSocialIcons.map(({ icon, label, href }) => ({
+    icon,
+    label,
+    href: href
+      .replace('currentUrl', encodeURIComponent(currentUrl))
+      .replace('postTitle', encodeURIComponent(postTitle)),
+  }))
+}
```

Comment:
If we are using `currentUrl` and `postTitle` as placeholders then it would be nice to extract them in variables.

```ts
const URL_PLACEHOLDER="currentUrl"
const TEXT_PLACEHOLDER="postTitle"

export const sharePostSocialIcons = [
  {
    icon: FacebookLogo,
    label: 'Facebook',
    href: `https://www.facebook.com/sharer/sharer.php?u=${URL_PLACEHOLDER}&quote=${TEXT_PLACEHOLDER}`,
  },
  // ...
]

export function getSharePostSocialIcons(currentUrl: string, postTitle: string) {
  return sharePostSocialIcons.map(({ icon, label, href }) => ({
    icon,
    label,
    href: href
      .replace(URL_PLACEHOLDER, encodeURIComponent(currentUrl))
      .replace(TEXT_PLACEHOLDER, encodeURIComponent(postTitle)),
  }))
}
```

Alternatively we could create individual functions to build shareable links for each platform, since they have different implementation. They would take the same arguments and return the string. For instance:

```ts
const FACEBOOK_BASE_SHARE_URL = 'https://www.facebook.com/sharer/sharer.php'

function generateFacebookShareLink(title: string) {
  const url = new URL(FACEBOOK_BASE_SHARE_URL)
  url.searchParams.append('u', BASE_URL)
  url.searchParams.append('quote', title)
  return url.toString()
}
```

Let me know what you think :)

Reply:
thank you, I like the first solution ❤️